preferencesfx/src/main/java/com/dlsc/preferencesfx/util/StorageHandler.java

```diff
@@ -1,243 +1,145 @@
-    String json = preferences.get(hash(breadcrumb), serializedDefault);
-    return FXCollections.observableArrayList(gson.fromJson(json, ArrayList.class));
-  }
-
-  /**
-   * Clears the preferences.
-   *
-   * @return true if successful, false if there was an exception.
-   */
-  public boolean clearPreferences() {
-    try {
-      preferences.clear();
-    } catch (BackingStoreException e) {
-      return false;
-    }
-    return true;
-  }
-
-  /**
-   * Generates a SHA-256 hash of a String.
-   * Since {@link Preferences#MAX_KEY_LENGTH} is 80, if the breadcrumb is over 80 characters, it
-   * will lead to an exception while saving. This method generates a SHA-256 hash of the breadcrumb
-   * to save / load as the key in {@link Preferences}, since those are guaranteed to be
-   * maximum 64 chars long.
-   *
-   * @return SHA-256 representation of breadcrumb
-   */
-  public String hash(String key) {
-    Objects.requireNonNull(key);
-    MessageDigest messageDigest = null;
-    try {
-      messageDigest = MessageDigest.getInstance("SHA-256");
-    } catch (NoSuchAlgorithmException e) {
-      LOGGER.error("Hashing algorithm not found!");
-    }
-    messageDigest.update(key.getBytes());
-    return new String(messageDigest.digest());
-  }
-
-  public Preferences getPreferences() {
-    return preferences;
-  }
 
+public interface StorageHandler {
+
+    /**
+     * Stores the last selected category in TreeSearchView.
+     *
+     * @param breadcrumb the category path as a breadcrumb string
+     */
+    void saveSelectedCategory(String breadcrumb);
+
+    /**
+     * Gets the last selected category in TreeSearchView.
+     *
+     * @return the breadcrumb string of the selected category. null if none is found
+     */
+    String loadSelectedCategory();
+
+    /**
+     * Stores the given divider position of the MasterDetailPane.
+     *
+     * @param dividerPosition the divider position to be stored
+     */
+    void saveDividerPosition(double dividerPosition);
+
+    /**
+     * Gets the stored divider position of the MasterDetailPane.
+     *
+     * @return the double value of the divider position. 0.2 if none is found
+     */
+    double loadDividerPosition();
+
+    /**
+     * Stores the window width of the PreferencesFxDialog.
+     *
+     * @param windowWidth the width of the window to be stored
+     */
+    void saveWindowWidth(double windowWidth);
+
+    /**
+     * Searches for the window width of the PreferencesFxDialog.
+     *
+     * @return the double value of the window width. 1000 if none is found
+     */
+    double loadWindowWidth();
+
+    /**
+     * Stores the window height of the PreferencesFxDialog.
+     *
+     * @param windowHeight the height of the window to be stored
+     */
+    void saveWindowHeight(double windowHeight);
+
+    /**
+     * Searches for the window height of the PreferencesFxDialog.
+     *
+     * @return the double value of the window height. 700 if none is found
```

Comment:
I would leave out the default value of the documentation of the interface, since this has to be decided by the implementor of the interface